### DISCUSSION TITLE
How to loose all your payara installations in one shot...

### DISCUSSION THREAD

**Discussion by roberto-cisternino** [apache/netbeans#6438] [General]
Using the latest Netbeans 19 I tried to install the latest Payara server and after selecting the root folder c:/Servers/payara I expected the install process will create a new folder payara-6.2023.8 but I obtained a completely different result... all my payara previous folder and subfolders (all my payara server installations) were deleted... no wiped for ever !!!
This behaviour is crazy and really dangerous, nothing that one can expect from NetBeans.

A disaster !!!!!!!

**Comment by mbien** [comment]
i can't reproduce this.

when I tried to overwrite a non-empty folder it backed it up, the code for that is here:
https://github.com/apache/netbeans/blob/f75da6ef22b2c637ee50ab10cc53a695a7c47a1e/enterprise/payara.common/src/org/netbeans/modules/payara/common/wizards/Retriever.java#L480-L499

**Comment by roberto-cisternino** [comment]
Well, in my case, in the payara root where all my different version were installed I found the new version but all the other versions were wiped (non simply canceled)

**Reply by roberto-cisternino** [reply]
The first attempt I selected c:/payara but that was not a valid folder according to Netebans... so I understood that I needed to provide a new folder under it e,g, payara-6.20023.8 and this worked out, but all c:/payara content desappeared except the new sub folder.
